src/test/java/com/pgs/client/supplier/AccessTokenSupplierTest.java

```diff
@@ -0,0 +1,40 @@
+package com.pgs.client.supplier;
+
+import com.pgs.client.dto.Token;
+import com.pgs.client.service.AuthenticationClient;
+import org.apache.commons.lang3.time.StopWatch;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class AccessTokenSupplierTest {
+
+    @Mock
+    private AuthenticationClient authenticationClient;
+    @Mock
+    private StopWatch stopWatch;
+    @InjectMocks
+    private AccessTokenSupplier accessTokenSupplier;
+
+    private static final Token TOKEN = Token.builder()
+            .accessToken("access1234")
+            .refreshToken("refresh1234")
+            .build();
+
+    @Test
+    void testSupplyToken() {
```

Comment:
AccessTokenSupplierTest has more logic so should have more test cases. At the same time AuthenticationInterceptor is a very simple component while it's test has multiple different test cases. 
I think you confused those 2 classes with each other in terms of unit tests.